core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -5290,8 +5290,10 @@ class KafkaApisTest {
 
     // read the header from the buffer first so that the body can be read next from the Request constructor
     val header = RequestHeader.parse(buffer)
-    val context = new RequestContext(header, "1", InetAddress.getLocalHost, KafkaPrincipal.ANONYMOUS,
-      listenerName, SecurityProtocol.PLAINTEXT, ClientInformation.EMPTY, fromPrivilegedListener,
+    // DelegationTokens require the context authenticated to be a
+    // non SecurityProtocol.PLAINTEXT and a non KafkaPrincipal.ANONYMOUS
+    val context = new RequestContext(header, "1", InetAddress.getLocalHost, new KafkaPrincipal(KafkaPrincipal.USER_TYPE, "Alice"),
+      listenerName, SecurityProtocol.SSL, ClientInformation.EMPTY, fromPrivilegedListener,
```

Comment:
A comment about how we test the failure case via integration tests (e.g. DelegationTokenRequestsOnPlainTextTest.scala) would be good to add so people don't think we missed testing that.

Reply:
Done